### ISSUE TITLE
Autocompleting environments ignores nesting

### ISSUE THREAD

**Issue by Focus** [Focus/latexer#15]
Example:

``` latex
\begin{something}
  \begin{something}%cursor here and press enter
\end{something}
```

this will not autocomplete. First reported in #12

**Comment by gasteigerjo** [comment]
Also, if you do this:

```
\begin{something} %cursor here and press enter
  \begin{something}
\end{something}
```

Latexer will insert another `\end{something}`, like this:

```
\begin{something}

\end{something}
  \begin{something}
\end{something}
```

which can be really annoying.

**Comment by Foggalong** [comment]
Yeah, this is especially annoying. Having this problem with `\begin{itemize}` and `\begin{enumerate}`

**Comment by Focus** [comment]
Fixed in 8a3ee17
